ddtrace/appsec/_iast/__init__.py

```diff
@@ -68,7 +73,22 @@ def ddtrace_iast_flask_patch():
     sys.modules[module_name] = compiled_code
 
 
+def enable_iast_propagation():
+    global _IS_IAST_REGISTERED
+    if asbool(os.getenv("DD_IAST_ENABLED", False)):
+        from ddtrace.appsec._iast._utils import _is_python_version_supported
+
+        if _is_python_version_supported():
+            from ddtrace.appsec._iast._ast.ast_patching import _should_iast_patch
+            from ddtrace.appsec._iast._loader import _exec_iast_patched_module
+
+            log.debug("IAST enabled")
+            ModuleWatchdog.register_pre_exec_module_hook(_should_iast_patch, _exec_iast_patched_module)
```

Comment:
I think this is answered below in a comment, but I'll still ask just to ensure I understand properly.

Reply:
Oops, `_IS_IAST_REGISTERED` is deprecated. I attempted an idempotent function with this global variable, but it was challenging to test and I considered it a potential bug generator.

Instead of that variable, I updated the `_pre_exec_module_hooks` list to a set to prevent duplicated hooks.